### PR TITLE
Update params builder

### PR DESCRIPTION
**Pull request by mynguyenlgg**

### CODE REVIEW

File: src/main/java/com/client/services/ParamsBuilder.java [hunk 1]
```diff
@@ -3,12 +3,33 @@
...
 import java.util.Map;
 
 @Setter
 @Getter
 @Builder
-public class ParamsBuilder<String, V> {
-    private Map<String, V> pathParams;
-    private Map<String, V> queryParams;
+public class ParamsBuilder<String, T> {
+    private Map<String, T> pathParams;
```
Comment (truongdnascript) [inline src/main/java/com/client/services/ParamsBuilder.java:14]: `private Map<String, T> pathParams = new HashMap<>();`

File: src/main/java/com/client/services/ParamsBuilder.java [hunk 2]
```diff
@@ -3,12 +3,33 @@
...
+    private Map<String, T> queryParams;
+
+    public ParamsBuilder(Map<String, T> pathParams, Map<String, T> queryParams) {
+        this.pathParams = pathParams != null ? new HashMap<>(pathParams) : new HashMap<>();
```
Comment (truongdnascript) [inline src/main/java/com/client/services/ParamsBuilder.java:18]: `this. pathParams=pathParams`

File: src/main/java/com/client/services/ParamsBuilder.java (hunk 1 above)
Comment (mynguyenlgg) [reply src/main/java/com/client/services/ParamsBuilder.java]: I updated.

File: src/main/java/com/client/services/ParamsBuilder.java (hunk 2 above)
Comment (mynguyenlgg) [reply src/main/java/com/client/services/ParamsBuilder.java]: I updated.

File: src/main/java/com/client/services/ParamsBuilder.java [hunk 3]
```diff
@@ -1,14 +1,32 @@
...
-    private Map<String, V> pathParams;
-    private Map<String, V> queryParams;
+@NoArgsConstructor
+@AllArgsConstructor
+public class ParamsBuilder<String, T> {
+    private Map<String, T> pathParams = new HashMap<>();
+    private Map<String, T> queryParams = new HashMap<>();
+
+    public ParamsBuilder<String, T> pathParam(String key, T value) {
+        if (this.pathParams != null) {
```
Comment (dinhbao10t4) [inline src/main/java/com/client/services/ParamsBuilder.java:20]: Do we need this condition? (if this.pathParams is null, the params will not put)

Comment (mynguyenlgg) [reply src/main/java/com/client/services/ParamsBuilder.java]: Yes, correct. I updated. Please help me review it again.

**Review by dinhbao10t4 — APPROVED** [review]

**Pull request merged**